### DISCUSSION TITLE
Drawing black with a theme set

### DISCUSSION THREAD

**Discussion by akhudek** [fltk-rs/fltk-rs#1393] [Q&A] [answered]
When I try to set a black background for a frame, or even if I override draw and raw a rectangle that is black, it seems to draw as white instead. I can draw Color::Black just fine however, it only seems to affect colors set with from_rgb. This happens if I apply a new color theme or widget theme and appears to only affect black. 

Is there a way to disable this behaviour? Here is a minimal example:

```rust
use fltk::{
    app,
    enums::{Color, FrameType},
    frame::Frame,
    prelude::*,
    window::Window,
};
use fltk_theme::{ThemeType, WidgetTheme};

fn main() {
    let app = app::App::default();
    let widget_theme = WidgetTheme::new(ThemeType::Dark);
    widget_theme.apply();

    let mut wind = Window::new(100, 100, 400, 300, "Hello from rust");
    let mut frame = Frame::new(10, 10, 50, 50, "");
    frame.set_color(Color::from_rgb(0, 0, 0));
    frame.set_frame(FrameType::FlatBox);

    wind.end();
    wind.show();
    app.run().unwrap();
}
```

**Comment by MoAlyousef** [comment, marked as answer]
Hi
Unfortunately no. See this discussion here:
https://www.reddit.com/r/FLTK/comments/12c3hgh/using_black_when_color_index_is_overwritten/

FLTK uses an RGBI color representation and has a color map. Changing the color map like when setting a color theme, causes RGB values which actually map into its colormap and have an indexing conflict (mainly the color black) would cause such behavior. 

You can either use Color::Black or Color::by_index(56) to avoid such conflict, with all (widget/color) themes except the SHAKE_THEME.
For the SHAKE_THEME, you would have to assign the color to some known color to FLTK like the Free color:
```rust
app::set_color(Color::Free, 0, 0, 0);
// * rest of code
frame.set_color(Color::Free);
```

**Comment by akhudek** [comment]
Ok thanks. I guess explicitly checking for black and replacing with Color::Black is then.
